lib/paper_trail/reifier.rb

```diff
@@ -219,6 +219,16 @@ def reify_has_many_through(transaction_id, associations, model, options = {})
             through_collection.each do |through_model|
               reify_has_manys(transaction_id, through_model, options)
             end
+
+            # At this point, the "through" part of the association chain has
+            # been reified, but not the final, "target" part. To continue our
+            # example, `model.sections` (including `model.sections.paragraphs`)
+            # has been loaded. However, the final "target" part of the
+            # association, that is, `model.paragraphs`, has not been loaded. So,
+            # we do that now.
+            collection = through_collection.map { |through_model|
+              through_model.public_send(assoc.name.to_sym).to_a
+            }.flatten
```

Comment:
`flat_map` is a slightly more efficient version that does both `map` and `flatten` in a single operation. It's more efficient because it doesn't have to create the intermediary array that's then discarded.